### ISSUE TITLE
Support for vector tiles

### ISSUE THREAD

**Issue by mvl22** [cyclestreets/android#110]
Vector tiles are where the web mapping community is going - rather than send a bitmap over the air/wire, vector data is sent, and the device renders this on the fly. The widely-used Gmaps app for instance does exactly this, requesting vector map data for the area and surrounding area as needed.

The major benefit of this approach is that map data appears quickly, especially in a slow connection area. The quality of the user experience is far superior.

We currently have basic support for this in the form of the mapsforge vector layer. But that requires a large download.

OpenCycleMap, the excellent cycle view we use, will be available in vector mapping at some point in the future and there are other vector tile sources emerging.

These rely on an OpenGL map renderer on the device. Background info:
http://gis.stackexchange.com/questions/64248/

The Android app ideally needs to support these, and there are a couple of libraries now available:

* http://mousebird.github.io/WhirlyGlobe/ (which is a full-scale map library, like Leaflet.js and mapsforge)
* https://www.mapbox.com/blog/android-gl-preview/ (though there are non-technical issues we would need to discuss before using this)

**Comment by jf1** [comment]
OsmAnd does an excellent job of rotating a pre-downloaded vector map as you cycle, so it's what I now use whenever my phone is handlebar- mounted. It'd be great if the Cyclestreets Android app could do similar in future. https://github.com/osmandapp/Osmand

**Comment by jf1** [comment]
Done. It was also a suggestion to look at their code which is released under GPLv3

**Comment by jezhiggins** [comment]
Martin, do we know the status of the OCM vector data?  Did that ever come to fruition?

**Comment by mvl22** [comment]
> do we know the status of the OCM vector data

Yes, I believe there is a possibility of getting the stylesheet. However, the problem is the rendering side.

However, I'd say a higher priority really is getting the existing vector download system working as well as possible, and the data updated and more areas covered - we get far more requests for that.
